operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -203,48 +203,97 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	// next we sync up all of our pools themselves
-	admin, requeue, err := r.reconcilePools(ctx, rp, pools)
+	requeue, err := r.reconcilePools(ctx, rp, pools)
 	if err != nil {
 		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
 
 		log.Error(err, "error reconciling pools")
 		return r.syncStatusErr(ctx, err, status, rp)
 	}
-	if requeue {
+	if requeue || !pools.AnyReady() {
+		// we might have no brokers ready at this point, so we can't do anything below this
+		// since they require the ability to talk to our brokers
 		return r.syncStatusAndRequeue(ctx, status, rp)
 	}
-	defer admin.Close()
-
-	status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonSynced)
 
-	version, requeue, err := r.reconcileClusterConfig(ctx, admin, rp)
+	admin, err := r.ClientFactory.RedpandaAdminClient(ctx, rp)
 	if err != nil {
-		status.Status.SetConfigurationApplied(statuses.ClusterConfigurationAppliedReasonError, err.Error())
-
-		log.Error(err, "error reconciling cluster config")
+		log.Error(err, "error fetching redpanda admin client")
 		return r.syncStatusErr(ctx, err, status, rp)
 	}
+	defer admin.Close()
 
-	status.ConfigVersion = ptr.To(version)
+	// now we ensure that we reconcile all of our decommissioning nodes
+	// TODO: Do we want to rate limit this as well given that it also calls the admin API?
+	// My thought is no since we want to be snappy with decommissioning.
+	health, requeue, err := r.reconcileDecommission(ctx, rp, admin, pools)
+	if err != nil {
+		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
+		if err != nil {
+			if internalclient.IsTerminalClientError(err) {
+				status.Status.SetHealthy(statuses.ClusterHealthyReasonTerminalError, err.Error())
+			} else {
+				status.Status.SetHealthy(statuses.ClusterHealthyReasonError, err.Error())
+			}
+		}
 
+		log.Error(err, "error decommissioning brokers")
+		return r.syncStatusErr(ctx, err, status, rp)
+	}
 	if requeue {
 		return r.syncStatusAndRequeue(ctx, status, rp)
 	}
 
-	status.Status.SetConfigurationApplied(statuses.ClusterConfigurationAppliedReasonApplied)
-
-	license, err := r.reconcileLicense(ctx, admin, rp, status)
-	if err != nil {
-		log.Error(err, "error reconciling license")
-		return r.syncStatusErr(ctx, err, status, rp)
+	status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonSynced)
+	if health.IsHealthy {
+		status.Status.SetHealthy(statuses.ClusterHealthyReasonHealthy)
+	} else {
+		// TODO: give more specific message here
+		status.Status.SetHealthy(statuses.ClusterHealthyReasonNotHealthy, "Cluster is not healthy")
 	}
-	if license != nil {
-		rp.Status.LicenseStatus = license
+
+	// we rate limit setting cluster configuration to once a minute if it's already been applied and is up-to-date
```

Comment:
Out of scope for this PR but I think we'd be better off putting this logic into the `reconcileClusterConfig` method as the primary reconcilation method is getting way to long and branch-y to easily reason about.

Maybe the pattern we shift to is `maybeReconcile____` and `reconcile____`?

Great comment!